core/chains/evm/client/client.go

```diff
@@ -48,9 +48,10 @@ type Client interface {
 
 	// HeadByNumber is a reimplemented version of HeaderByNumber due to a
```

Comment:
nit: Edit comment to say both HeadByNumber and HeadByHash are a modified reimplementation

Reply:
Actually it will also be great if you can explain in the comments here what exactly was the issue in Avalanche.
Because if later on someone looks at this code, they won't know what exactly the incompatibility with Avalanche was, and thus not be able to refactor this code without actually tracking down the Avalanche issue.

Reply:
I think we should just remove the HeaderByXX methods entirely to avoid this footgun. Wdyt?

Reply:
Yes, makes more sense to remove HeaderByXX